runtime/onert/core/src/util/shapeinf/Select.cc

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape MostElement(const ir::Shape &input_cond_shape, const ir::Shape &input_true_shape,
+                      const ir::Shape &input_false_shape)
+{
+  uint64_t num_cond = input_cond_shape.num_elements();
+  uint64_t num_true = input_true_shape.num_elements();
+  uint64_t num_false = input_false_shape.num_elements();
+
+  ir::Shape most_shape = (num_cond >= num_true) && (num_cond >= num_false)
+                             ? input_cond_shape
+                             : (num_false >= num_true ? input_false_shape : input_true_shape);
+
+  return most_shape;
+}
+
+ir::Shape inferSelectShape(const ir::Shape &input_cond_shape, const ir::Shape &input_true_shape,
+                           const ir::Shape &input_false_shape)
+{
+  // Change to the shape with the most_elements by broadcasting
+  ir::Shape most_Shape = MostElement(input_cond_shape, input_true_shape, input_false_shape);
```

Comment:
I don't understand why you did this.

In case of
`
input_cond_shape = {1}
input_true_shape = {1, 2}
input_false_shape = {2, 1},
`
output_shape becomes `{2, 2}`.

But in this logic it becomes `{2, 1}`.
Please check more.

Reply:
I agree with @YiHyunJin . And please throw exception when shape cannot broadcast. (ex. `true_shape = {2, 3}`, `false_shape = {3, 2}`)

Reply:
I will modify